src/scripting/game_lua_kernel.cpp

```diff
@@ -2784,173 +2849,74 @@ int game_lua_kernel::intf_match_side(lua_State *L)
 	return 1;
 }
 
-int game_lua_kernel::intf_modify_ai_wml(lua_State *L)
+int game_lua_kernel::intf_set_team_id(lua_State *L)
 {
-	vconfig cfg(luaW_checkvconfig(L, 1));
+	int team_i = luaL_checkinteger(L, 1) - 1;
+	std::string flag = luaL_optlstring(L, 2, "", nullptr);
+	std::string color = luaL_optlstring(L, 3, "", nullptr);
 
-	side_filter ssf(cfg, &game_state_);
-	std::vector<int> sides = ssf.get_teams();
-	for (const int &side_num : sides)
-	{
-		ai::manager::modify_active_ai_for_side(side_num,cfg.get_parsed_config());
+	if(flag.empty() && color.empty()) {
+		return 0;
+	}
+	team& side = teams()[team_i];
```

Comment:
This looks like it misses need a check wheterh team_i is actuall in range of teams().

Also i think this functions name is missleading in 2 ways:
1) All the other functions use teh tern 'side' (match_side, get_sides...)
2) It doesnt chage an 'id' (side= or save_id=) of that side at all.